chain/client/src/client.rs

```diff
@@ -960,6 +985,18 @@ impl Client {
         Ok(())
     }
 
+    /// Gets the advanced timestamp delta in nanoseconds for sandbox once it has been fast-forwarded
+    #[cfg(feature = "sandbox")]
+    pub fn sandbox_delta_time(&self) -> u64 {
+        let avg_block_prod_time = (self.config.min_block_production_delay.as_nanos()
+            + self.config.max_block_production_delay.as_nanos())
+            / 2;
```

Comment:
No this is not representative of average block time. What do you need it for?

Reply:
I'm using it for estimating how much of a jump in the timestamp needs to be added to the current time, for the new timestamp after fast-forwarding/time-travelling. It doesn't have to be entirely accurate. Just need to be within a threshold. Is `min_block_production_delay` and `max_block_production_delay` not the right config to use then? It seemed so far to me to have controlled how fast block production while I was testing it when I was playing with it

Reply:
> I'm using it for estimating how much of a jump in the timestamp needs to be added to the current time, for the new timestamp after fast-forwarding/time-travelling

What do you use this estimation for?

Reply:
after we fast-forward the sandbox by a specific delta height, we need the new blocks produced to reflect the timestamps at that new block height as well. So this estimation is done specifically for that.

Reply:
> we need the new blocks produced to reflect the timestamps at that new block height as well

I am confused by this statement. Which of the following do you mean here:
1. Sandbox assumes a certain relationship between block height and block timestamp and it is crucial for the functionality of Sandbox that such relationship holds.
2. Block height is used as a proxy for block timestamp so that you only need to update one value to make fast forwarding work

Reply:
More so 1, but 2 also holds as a side-effect of how I implemented it. The functionality I'm trying to simulate here is as if the blocks were produced as normal. So let's say we fast-forward 10,000 blocks. That translates to producing 10,000 blocks where the latest block timestamp should reflect an almost equivalent amount of time has passed after 10,000 blocks were produced. So roughly the new timestamp should be `time::now() + 10,000 * avg_block_prod_time`.

Let me know if that doesn't make sense since this feature is more so based on what the sdk-sim had with producing blocks to simulate fast-forwarding/time-traveling

Reply:
I see. What would break in sandbox if the time isn't fast-forwarded? In other words, is this a nice-to-have feature or does it serve some crucial purpose?

Reply:
For instance, the cron.cat needs it to test out scheduling in their contracts. So I would vote for crucial purpose. @TrevorTJClarke can talk more about it from his side and how he uses it